core/ast/src/source.rs

```diff
@@ -91,6 +92,11 @@ impl Script {
 
         Ok(bindings)
     }
+
+    /// Takes the source text.
+    pub fn take_source(&mut self) -> Option<SourceText> {
+        self.source.take()
+    }
```

Comment:
The AST should be mostly immutable after parsing so I would change this to returning `&SourceText`. This would also be a problem if you would try to compile a `Script` multiple times.